android/java/org/chromium/chrome/browser/playlist/hls_content/HlsUtils.java

```diff
@@ -0,0 +1,187 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+package org.chromium.chrome.browser.playlist.hls_content;
+
+import android.content.Context;
+
+import androidx.media3.common.util.UriUtil;
+import androidx.media3.exoplayer.hls.playlist.HlsMediaPlaylist.Segment;
+
+import com.brave.playlist.util.HLSParsingUtil;
+import com.brave.playlist.util.MediaUtils;
+
+import org.chromium.base.PathUtils;
+import org.chromium.chrome.browser.playlist.PlaylistStreamingObserverImpl;
+import org.chromium.playlist.mojom.PlaylistItem;
+import org.chromium.playlist.mojom.PlaylistService;
+
+import java.io.File;
+import java.util.Queue;
+
+public class HlsUtils {
+    private static final String GET_METHOD = "GET";
+    private static final String TAG = "Playlist.HlsUtils";
+
+    public interface HlsManifestDelegate {
+        default void onHlsManifestCompleted(Queue<Segment> segmentsQueue) {}
+    }
+
+    public interface HlsFileDelegate {
+        default void onProgress(int sofar) {}
+        default void onReady(String filePath) {}
+    }
+
+    private static int sSofar;
+    private static PlaylistStreamingObserverImpl sPlaylistStreamingObserverImpl;
+
+    public static void getManifestFile(Context context, PlaylistService playlistService,
+            PlaylistItem playlistItem, HlsUtils.HlsManifestDelegate hlsManifestDelegate) {
+        String mediaPath = getHlsMediaFilePath(playlistItem);
+        String hlsManifestFilePath = getHlsManifestFilePath(playlistItem);
+        final String manifestUrl = getHlsResolutionManifestUrl(context, playlistItem);
+        if (playlistService != null) {
+            playlistService.requestStreamingQuery(manifestUrl, GET_METHOD);
+            PlaylistStreamingObserverImpl
+                    .PlaylistStreamingObserverImplDelegate playlistStreamingObserverImplDelegate =
+                    new PlaylistStreamingObserverImpl.PlaylistStreamingObserverImplDelegate() {
+                        @Override
+                        public void onResponseStarted(String url, long contentLength) {
+                            try {
+                                deleteFileIfExist(hlsManifestFilePath);
+                            } catch (Exception ex) {
+                                ex.printStackTrace();
```

Comment:
do we want a good formed logs here and in a diff places in that class instead of just a stack trace?